plugins/frontend/backstage-plugin-cloudsmith/src/api/CloudsmithClient.ts

```diff
@@ -53,4 +67,54 @@ export class CloudsmithClient implements CloudsmithApi {
     }
     return await response.json();
   }
+
+  // get data from Cloudsmith quota endpoint (https://help.cloudsmith.io/reference/quota_read)
+  async getQuota({ owner }: { owner: string }): Promise<cloudsmithUsage> {
+    const cloudsmithApiUrl = await this.getApiUrl();
+    const response = await this.fetchApi.fetch(
+      `${cloudsmithApiUrl}/quota/${owner}/`,
+    );
+    if (!response.ok) {
```

Comment:
Can/should we add explicit checking for the paywall like this, and then in the component specifically handle for this error providing the user a link to the cloudsmith marketting website or a link instructing them what to do when they get this error?

if (response.status === 402) {
  throw new PaymentRequiredError((await response.json()).detail)
}

Reply:
Maybe this warrents a little reuse, because the other functions would require the same handling, creating a little bit of duplication.